subsystems/motor/speedmodifiers/AccelerationCap.java

```diff
@@ -61,28 +136,8 @@ public AccelerationCap(PDP pdp) {
 	 */
 	@Override
 	public double modify(double inputSpeed) {
-		double outputSpeed;
-		if (Math.abs(inputSpeed) > Math.abs(currentSpeed) && pdp.getVoltage() < softStopVoltage) {
-			LogKitten.d("AccelerationCap brownout protecting");
-			if (pdp.getVoltage() < hardStopVoltage) {
-				outputSpeed = currentSpeed - 0.3 * currentSpeed;
-			} else {
-				outputSpeed = currentSpeed;
-			}
-			LogKitten.d("AccelerationCap input: " + inputSpeed + "\t" + "AccelerationCap output: " + outputSpeed);
-		} else if (Math.abs(inputSpeed) > Math.abs(currentSpeed)) {
-			LogKitten.d("AccelerationCap voltage ramping");
-			outputSpeed = currentSpeed
-				+ ((double) (System.currentTimeMillis() - lastUpdate) / 64) * (inputSpeed - currentSpeed);
-			if (outputSpeed > inputSpeed) {
-				outputSpeed = inputSpeed;
-			}
-		} else {
-			outputSpeed = inputSpeed;
-		}
-		LogKitten.d("AccelerationCap input: " + inputSpeed + "\t" + "AccelerationCap output: " + outputSpeed);
-		lastUpdate = System.currentTimeMillis();
-		currentSpeed = outputSpeed;
-		return outputSpeed;
+		currentSpeed = calculate(inputSpeed);
```

Comment:
This allows for multiple exit points from the calculation while still storing currentSpeed.